### PR TITLE
Fixed typeguard error by widening type scope to Any #825

### LINKED ISSUE #825: GroupBy.nunique() fails on Categorical.from_codes()
I ran into the following bug while working Issue #692 
```python       
cat = ak.Categorical.from_codes(codes=ak.array([0, 1, 0, 1, 2]), categories=ak.array(['a', 'b', 'c']))
i = ak.arange(cat.size)

cat_group = ak.GroupBy(cat)
cat_labels, cat_values = cat_group.nunique(i)
```
calling nunique on a `Categorical.from_codes()` results in a `TypeError`

This only seems to impact `Categorical`s made using `.from_codes()` because the following runs fine
```python
cat = ak.Categorical(ak.array(['a', 'b', 'a', 'b', 'c']))
i = ak.arange(cat.size)

cat_group = ak.GroupBy(cat)
cat_labels, cat_values = cat_group.nunique(i)
```

This seems similar to Issue #777, so I expect this is the result of overly restrictive typing in `aggregate`

### PR DESCRIPTION
**Pull request by hokiegeek2**
This PR does the following:

1. Fixes #825 by widening the scope of nunique and aggregate return[0] element to Any
2. Adds aggregate unit tests to confirm bug fix

### CODE REVIEW

**Review by pierce314159 — APPROVED** [review]
Looks great thanks @hokiegeek2!

**Review by glitch — APPROVED** [review]

File: arkouda/groupbyclass.py [hunk 1]
```diff
@@ -639,8 +639,7 @@ def argmax(self, values : pdarray)\
             raise TypeError('argmax is only supported for pdarrays of dtype float64 and int64')
         return self.aggregate(values, "argmax")
     
-    def nunique(self, values : pdarray) \
-                    -> Tuple[Union[pdarray,List[Union[pdarray,Strings]]],pdarray]:
+    def nunique(self, values : pdarray) -> Tuple[Any,pdarray]:
```
Comment (reuster986) [inline arkouda/groupbyclass.py:642]: Should we apply this change to all the other reduction methods (`.sum()`, `.min()`, `.max()`, etc.)? I'm thinking they will encounter the same problem as `.nunique()` with GroupBy on Strings or Categorical.

**Review by reuster986 — COMMENTED** [review]
I definitely like this change, I'm just wondering if it should be applied to all the `GroupBy` methods, not just `.nunique()`. Thoughts?

File: arkouda/groupbyclass.py (hunk 1 above)
Comment (hokiegeek2) [reply arkouda/groupbyclass.py:642]: @reuster986 Good point and I vote yes

Comment (hokiegeek2) [reply arkouda/groupbyclass.py:642]: I recommend a separate one so we can get this merged into master for @pierce314159 work that is dependent upon this bug fix

**Pull request merged**